internal/consensus/reactor.go

```diff
@@ -1284,11 +1284,12 @@ func (r *Reactor) handleMessage(ctx context.Context, envelope *p2p.Envelope, cha
 	// and because a large part of the core business logic depends on these
 	// domain types opposed to simply working with the Proto types.
 	protoMsg := new(tmcons.Message)
-	if err := protoMsg.Wrap(envelope.Message); err != nil {
+	if err = protoMsg.Wrap(envelope.Message); err != nil {
```

Comment:
You could (if you prefer) leave these local—whatever `return` returns will get assigned to the named return value on the way out the gate.

Reply:
https://go.dev/play/p/X0s-9fN3O9N

Reply:
the named return variables are always gross. 

hopefully this all can get better in other ways, with panic handling somewhere else?

Reply:
> https://go.dev/play/p/X0s-9fN3O9N

I would have never guessed that this worked that way.

Reply:
Yeah, I try to avoid using named returns unless they provide useful documentation, or there's a defer (which obliges me, as in this case).

There are only a couple "real" use cases for naked returns, and I agree with you completely that they're best avoided even if you need the named return values for other reasons.